### PR TITLE
Time formats

### PR DESCRIPTION
**Pull request by ruthmd**
Timestamp validation with seconds, milliseconds  and microseconds

### CODE REVIEW

File: j2v/utils/helpers.py [hunk 1]
```diff
@@ -53,13 +53,24 @@ def is_ISO_Time(dim_val):
...
+    try:
+        timestamp_now = datetime.timestamp(datetime.now())
+        timestamp_len = len(str(dim_val))
+        if dim_val > 0:
+            if timestamp_len == 13:
+                dim_val = dim_val/10**3
+            elif timestamp_len == 16:
+                dim_val = dim_val / 10 ** 6
+            # check if modifications of dim_val can generate valid dates
+            map(datetime.fromtimestamp, [dim_val, timestamp_now - 3.154e+7, timestamp_now + 3.154e+7])
```
Comment (thoean) [inline j2v/utils/helpers.py]: Can you comment why `3.154e+7`?

**Comment by mzas** [comment]
@ruthmd Please merge today's new unit tests to this branch. Even without new tests, one test is failing now. Please run tests before creating PRs and make sure they are passed.

**Comment by thoean** [comment]
> @ruthmd Please merge today's new unit tests to this branch. Even without new tests, one test is failing now. Please run tests before creating PRs and make sure they are passed.

It'd be good to have an automated build step at least for the unit tests at some point. GitHub also does the right thing by auto-merging the target branch into the current branch before building it to avoid such surprises like the one you mentioned.

**Comment by mzas** [comment]
> > @ruthmd Please merge today's new unit tests to this branch. Even without new tests, one test is failing now. Please run tests before creating PRs and make sure they are passed.
> 
> It'd be good to have an automated build step at least for the unit tests at some point. GitHub also does the right thing by auto-merging the target branch into the current branch before building it to avoid such surprises like the one you mentioned.

Good point, I will set up the automated CI.

File: j2v/utils/helpers.py [hunk 2]
```diff
@@ -53,13 +53,24 @@ def is_ISO_Time(dim_val):
...
+    try:
+        timestamp_now = datetime.timestamp(datetime.now())
+        timestamp_len = len(str(dim_val))
+        if dim_val > 0:
+            if timestamp_len == 13:
+                is_unix_timestamp(dim_val/10**3)
+            elif timestamp_len == 16:
+                is_unix_timestamp(dim_val/10**6)
+            elif timestamp_len == 10:
+                map(datetime.fromtimestamp, [dim_val, timestamp_now - 3.154e+7, timestamp_now + 3.154e+7])
```
Comment (mzas) [inline j2v/utils/helpers.py]: 1. This check will be never executed. To understand why, try the following in your python console:
`def foo(x):
    raise Exception`
`foo(1)`
Then do what you are trying to achieve here:
`map(foo,[1,2,3,4])`
What's the learning?
2. I don't get this numbers `timestamp_now - 3.154e+7, timestamp_now + 3.154e+7`, what's the intent of checking today-1y or today+1y?

**Comment by mzas** [comment]
I reviewed the changes, the problems are fixed.

**Pull request merged**